src/bms/parse/random/ast_build.rs

```diff
@@ -106,18 +107,19 @@ pub(super) fn build_control_flow_ast<'a>(
 /// Handle a single Token: if it is the start of a block, recursively call the block parser, otherwise return a Token node.
 fn parse_unit_or_block<'a>(
     iter: &mut BmsParseTokenIter<'a>,
-) -> Option<(Unit<'a>, Vec<ControlFlowRule>)> {
+) -> Option<(Unit<'a>, Vec<ParseWarning>)> {
     let token = iter.peek()?;
-    match token {
-        Token::SetSwitch(_) | Token::Switch(_) => {
+    use TokenContent::*;
+    match &token.content {
+        SetSwitch(_) | Switch(_) => {
             let (unit, errs) = parse_switch_block(iter);
             Some((unit, errs))
         }
-        Token::Random(_) | Token::SetRandom(_) => {
+        Random(_) | SetRandom(_) => {
             let (unit, errs) = parse_random_block(iter);
             Some((unit, errs))
         }
-        token if !token.is_control_flow_token() => {
+        _ if !token.content.is_control_flow_token() => {
```

Comment:
```suggestion
        content if !content.is_control_flow_token() => {
```
It can shorten with the discarded one.